File: frontend/src/types/itemprocs.ts
export interface ItemProcInput {
  itemproc_input_id: number
  input_no: number
  item_id: number
  item_cd: string
  item_nm: string
  from_location_id: number | null
  from_location_cd: string
  from_location_nm: string
  req_qty: string | number | null
}

export interface ItemProc {
  itemproc_id: number
  line_no: number
  wip_location_id: number
  wip_location_cd: string
  wip_location_nm: string
  output_item_id: number
  output_item_cd: string
  output_item_nm: string
  inputs: ItemProcInput[]
  created_at?: string | null
  updated_at?: string | null
}

export interface ItemProcessesOut {
  item_id: number
  item_cd: string
  item_nm: string
  processes: ItemProc[]
}

export interface ItemProcInputWrite {
  input_no: number
  item_id: number
  req_qty: number | null
}

export interface ItemProcWrite {
  line_no: number
  wip_location_id: number
  output_item_id: number
  inputs: ItemProcInputWrite[]
}

export interface ItemProcessesSave {
  processes: ItemProcWrite[]
}

export interface ItemProcessFinalItem {
  item_id: number
  item_cd: string
  item_nm: string
  itemtyp_cd: string
  customer_cd: string
}

export interface ItemProcessFinalItemsSave {
  item_ids: number[]
}

File: backend/app/schemas/itemprocs.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ItemProcInputWrite(BaseModel):
    input_no: int = Field(ge=1)
    item_id: int = Field(gt=0)
    from_location_id: int = Field(gt=0)
    req_qty: Decimal = Field(gt=0)


class ItemProcInputRead(BaseModel):
    itemproc_input_id: int
    input_no: int
    item_id: int
    item_cd: str
    item_nm: str
    from_location_id: int
    from_location_cd: str
    from_location_nm: str
    req_qty: Decimal


class ItemProcWrite(BaseModel):
    line_no: int = Field(ge=1)
    wip_location_id: int = Field(gt=0)
    rm_location_id: int = Field(gt=0)
    output_item_id: int = Field(gt=0)
    inputs: list[ItemProcInputWrite] = Field(default_factory=list)


class ItemProcRead(BaseModel):
    itemproc_id: int
    line_no: int
    wip_location_id: int
    wip_location_cd: str
    wip_location_nm: str
    rm_location_id: int
    rm_location_cd: str
    rm_location_nm: str
    output_item_id: int
    output_item_cd: str
    output_item_nm: str
    inputs: list[ItemProcInputRead]
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ItemProcessesOut(BaseModel):
    item_id: int
    item_cd: str
    item_nm: str
    processes: list[ItemProcRead]


class ItemProcessesSave(BaseModel):
    processes: list[ItemProcWrite] = Field(default_factory=list)


class ItemProcessFinalItemRead(BaseModel):
    item_id: int
    item_cd: str
    item_nm: str
    itemtyp_cd: str
    customer_cd: str = ""
